File: clients/java/src/main/java/com/svix/coyote/apis/Cache.java
// this file is @generated
package com.svix.coyote.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.coyote.ApiException;
import com.svix.coyote.HttpClient;
import com.svix.coyote.Utils;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import okhttp3.Headers;
import okhttp3.HttpUrl;
import com.svix.coyote.models.CacheDeleteIn;
import com.svix.coyote.models.CacheDeleteOut;
import com.svix.coyote.models.CacheGetIn;
import com.svix.coyote.models.CacheGetNamespaceIn;
import com.svix.coyote.models.CacheGetNamespaceOut;
import com.svix.coyote.models.CacheGetOut;
import com.svix.coyote.models.CacheSetIn;
import com.svix.coyote.models.CacheSetOut;

public class Cache {
    private final HttpClient client;

    public Cache(HttpClient client) {
        this.client = client;
    }

    /** Cache Set */
    public CacheSetOut set(
        final CacheSetIn cacheSetIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/cache/set");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            cacheSetIn,
            CacheSetOut.class
            );
    }

    /** Cache Get */
    public CacheGetOut get(
        final CacheGetIn cacheGetIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/cache/get");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            cacheGetIn,
            CacheGetOut.class
            );
    }

    /** Get cache namespace */
    public CacheGetNamespaceOut getNamespace(
        final CacheGetNamespaceIn cacheGetNamespaceIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/cache/get-namespace");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            cacheGetNamespaceIn,
            CacheGetNamespaceOut.class
            );
    }

    /** Cache Delete */
    public CacheDeleteOut delete(
        final CacheDeleteIn cacheDeleteIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/cache/delete");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            cacheDeleteIn,
            CacheDeleteOut.class
            );
    }
}

File: clients/python/coyote/apis/cache.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    CacheDeleteIn,
    CacheDeleteOut,
    CacheGetIn,
    CacheGetNamespaceIn,
    CacheGetNamespaceOut,
    CacheGetOut,
    CacheSetIn,
    CacheSetOut,
)


class CacheAsync(ApiBase):
    async def set(
        self,
        cache_set_in: CacheSetIn,
    ) -> CacheSetOut:
        """Cache Set"""
        body = cache_set_in.model_dump(exclude_unset=True, by_alias=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/cache/set",
            body=body,
            response_type=CacheSetOut,
        )

    async def get(
        self,
        cache_get_in: CacheGetIn,
    ) -> CacheGetOut:
        """Cache Get"""
        body = cache_get_in.model_dump(exclude_unset=True, by_alias=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/cache/get",
            body=body,
            response_type=CacheGetOut,
        )

    async def get_namespace(
        self,
        cache_get_namespace_in: CacheGetNamespaceIn,
    ) -> CacheGetNamespaceOut:
        """Get cache namespace"""
        body = cache_get_namespace_in.model_dump(exclude_unset=True, by_alias=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/cache/get-namespace",
            body=body,
            response_type=CacheGetNamespaceOut,
        )

    async def delete(
        self,
        cache_delete_in: CacheDeleteIn,
    ) -> CacheDeleteOut:
        """Cache Delete"""
        body = cache_delete_in.model_dump(exclude_unset=True, by_alias=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/cache/delete",
            body=body,
            response_type=CacheDeleteOut,
        )


class Cache(ApiBase):
    def set(
        self,
        cache_set_in: CacheSetIn,
    ) -> CacheSetOut:
        """Cache Set"""
        body = cache_set_in.model_dump(exclude_unset=True, by_alias=True)

        return self._request_sync(
            method="post",
            path="/api/v1/cache/set",
            body=body,
            response_type=CacheSetOut,
        )

    def get(
        self,
        cache_get_in: CacheGetIn,
    ) -> CacheGetOut:
        """Cache Get"""
        body = cache_get_in.model_dump(exclude_unset=True, by_alias=True)

        return self._request_sync(
            method="post",
            path="/api/v1/cache/get",
            body=body,
            response_type=CacheGetOut,
        )

    def get_namespace(
        self,
        cache_get_namespace_in: CacheGetNamespaceIn,
    ) -> CacheGetNamespaceOut:
        """Get cache namespace"""
        body = cache_get_namespace_in.model_dump(exclude_unset=True, by_alias=True)

        return self._request_sync(
            method="post",
            path="/api/v1/cache/get-namespace",
            body=body,
            response_type=CacheGetNamespaceOut,
        )

    def delete(
        self,
        cache_delete_in: CacheDeleteIn,
    ) -> CacheDeleteOut:
        """Cache Delete"""
        body = cache_delete_in.model_dump(exclude_unset=True, by_alias=True)

        return self._request_sync(
            method="post",
            path="/api/v1/cache/delete",
            body=body,
            response_type=CacheDeleteOut,
        )
